models/models/billing.py

```diff
@@ -1,135 +1,75 @@
-    sequencing_type: str | None
-    stage: str | None
-    sequencing_group: str | None
-
-    export_time: datetime.datetime | None
-    cost: str | None
-    currency: str | None
-    currency_conversion_rate: str | None
-    invoice_month: str | None
-    cost_type: str | None
-
-    class Config:
-        """Config for BillingRowRecord Response"""
-
-        orm_mode = True
+    cost_category: str | None
+    cost: float | None
+    day: datetime.date | None
 
     @staticmethod
-    def from_json(record):
-        """Create BillingRowRecord from json"""
-
-        record['service'] = record['service'] if record['service'] else {}
-        record['project'] = record['project'] if record['project'] else {}
-        record['invoice'] = record['invoice'] if record['invoice'] else {}
-        record['sku'] = record['sku'] if record['sku'] else {}
-
-        labels = {}
-
-        if record['labels']:
-            for lbl in record['labels']:
-                labels[lbl['key']] = lbl['value']
-
-        record['labels'] = labels
-
-        return BillingRowRecord(
-            id=record['id'],
-            topic=record['topic'],
-            service_id=record['service'].get('id'),
-            service_description=record['service'].get('description'),
-            sku_id=record['sku'].get('id'),
-            sku_description=record['sku'].get('description'),
-            usage_start_time=record['usage_start_time'],
-            usage_end_time=record['usage_end_time'],
-            gcp_project_id=record['project'].get('id'),
-            gcp_project_number=record['project'].get('number'),
-            gcp_project_name=record['project'].get('name'),
-            # labels
-            dataset=record['labels'].get('dataset'),
-            batch_id=record['labels'].get('batch_id'),
-            job_id=record['labels'].get('job_id'),
-            batch_name=record['labels'].get('batch_name'),
-            sequencing_type=record['labels'].get('sequencing_type'),
-            stage=record['labels'].get('stage'),
-            sequencing_group=record['labels'].get('sequencing_group'),
-            export_time=record['export_time'],
-            cost=record['cost'],
-            currency=record['currency'],
-            currency_conversion_rate=record['currency_conversion_rate'],
-            invoice_month=record['invoice'].get('month', ''),
-            cost_type=record['cost_type'],
+    def from_db(**kwargs):
+        """
+        Convert from db keys, mainly converting id to id_
+        """
+        return BillingInternal(
+            id=kwargs.get('id'),
+            ar_guid=kwargs.get('ar_guid', kwargs.get('ar-guid')),
+            gcp_project=kwargs.get('gcp_project'),
+            topic=kwargs.get('topic'),
+            batch_id=kwargs.get('batch_id'),
+            cost_category=kwargs.get('cost_category'),
+            cost=kwargs.get('cost'),
+            day=kwargs.get('day'),
         )
 
 
 class BillingColumn(str, Enum):
```

Comment:
Should this be in enums?

Reply:
I know, we started with enum, then added bunch of functions later. We can revisit later.